tests/versioned/koa-router.tap.js

```diff
@@ -123,6 +123,19 @@ tap.test('koa-router instrumentation', (t) => {
       run()
     })
 
+    t.test('should handle transaction state loss', (t) => {
+      let savedCtx = null
+      router.get('/:any', (ctx) => {
+        savedCtx = ctx
+      })
+      app.use(router.middleware())
+      helper.agent.on('transactionFinished', () => {
+        t.doesNotThrow(() => savedCtx._matchedRoute = 'test')
+        t.end()
```

Comment:
I see here we are simulating transaction state loss by writing to `_matchedRoute` once the transaction has completed.